### PR TITLE
Added BLE connect/disconnect events and fixed bug in MagnetometerServ…

### PR DESCRIPTION
**Pull request by bluetooth-mdw**
…ice re: bearing characteristic

See https://drive.google.com/file/d/0B9Z54U0dfwbFN0Y0RVVROS1DQWM/view?usp=sharing for testing of BLE connection events.

### CODE REVIEW

File: inc/bluetooth/ExternalEvents.h [hunk 1]
```diff
@@ -33,6 +33,9 @@ DEALINGS IN THE SOFTWARE.
 #define MICROBIT_ID_BLE             1000
 #define MICROBIT_ID_BLE_UART        1001
 
+#define MICROBIT_BLE_CONNECTED      1
```
Comment (jamesadevine) [inline inc/bluetooth/ExternalEvents.h]: Hi @bluetooth-mdw,

Great PR, just a few minor things

We have an unofficial connection in terms of event codes, for example:
`MICROBIT_COMPASS_EVT_CAL_END`

If both `MICROBIT_BLE_CONNECTED` and `MICROBIT_BLE_DISCONNECTED ` could be renamed that would be great!

File: source/bluetooth/MicroBitBLEManager.cpp [hunk 2]
```diff
@@ -267,6 +277,9 @@ void MicroBitBLEManager::init(ManagedString deviceName, ManagedString serialNumb
     // automatically restart advertising after a device disconnects.
     ble->gap().onDisconnection(bleDisconnectionCallback);
     ble->gattServer().onSysAttrMissing(bleSysAttrMissingCallback);
+    
+    // generate an event when a Bluetooth connection is lost
```
Comment (jamesadevine) [inline source/bluetooth/MicroBitBLEManager.cpp]: I think this comment should read:

"generate an event when a Bluetooth device connects" or something along those lines.

Comment (remay) [inline source/bluetooth/MicroBitBLEManager.cpp]: // generate an event when a Bluetooth connection is **established**

Comment (jamesadevine) [inline source/bluetooth/MicroBitBLEManager.cpp]: @remay 's version is a lot better... :smile:

**Comment by jamesadevine** [comment]
@bluetooth-mdw Am I right in thinking this will fix #114 as well?

**Comment by bluetooth-mdw** [comment]
Hopefully https://github.com/lancaster-university/microbit-dal/issues/114 will be fixed too. I've not witnessed this issue though so maybe the issue reporter can re-test?

Only just realised there's a comment here about a code comment! Let me get on that too.

**Comment by remay** [comment]
@jamesadevine I was going to point out that we change the BLE disconect handler when we reset() the device (see #29).  But I either can't find my way around the new code, or we've lost this functionality.  Can you point me in the right direction?

**Comment by jamesadevine** [comment]
@remay great catch... https://github.com/lancaster-university/microbit/blob/master/inc/MicroBit.h#L418

Martin - could you also make a PR here as well?

**Comment by jamesadevine** [comment]
@bluetooth-mdw - this looks good to me!

**Comment by bluetooth-mdw** [comment]
My first contribution to the micro:bit runtime. The world will never be the same again! :-)

**Comment by remay** [comment]
@jamesadevine @bluetooth-mdw 

> great catch... https://github.com/lancaster-university/microbit/blob/master/inc/MicroBit.h#L418

This implementation doesn't change the disconnect callback handler, so nothing to do.

**Comment by jamesadevine** [comment]
@remay very true - thanks @bluetooth-mdw

**Pull request merged**